monero/wallet_client.go

```diff
@@ -135,14 +172,53 @@ func (c *walletClient) GetBalance(idx uint64) (*wallet.GetBalanceResponse, error
 	})
 }
 
+// WaitForTransReceipt waits for the passed monero transaction ID to receive
+// numConfirmations and returns the transfer information. While this function will always
+// wait for the transaction to leave the mem-pool even if zero confirmations are
+// requested, it is the caller's responsibility to request enough confirmations that the
+// returned transfer information will not be invalidated by a block reorg.
+func (c *walletClient) WaitForTransReceipt(req *WaitForReceiptRequest) (*wallet.Transfer, error) {
+	height, err := c.GetHeight()
+	if err != nil {
+		return nil, err
+	}
+
+	var transfer *wallet.Transfer
+
+	for {
+		transferResp, err := c.rpc.GetTransferByTxid(&wallet.GetTransferByTxidRequest{
+			TxID:         req.TxID,
+			AccountIndex: 0,
+		})
+		if err != nil {
+			return nil, err
+		}
+		transfer = &transferResp.Transfer
+		// wait for transaction be mined (height set) even if 0 confirmations requested
+		if transfer.Height > 0 && transfer.Confirmations >= req.NumConfirmations {
+			break
+		}
+		log.Infof("Received %d of %d confirmations of XMR TXID=%s (height=%d)",
```

Comment:
move the log above the if statement (so it logs the exit case?)

Reply:
I'll do that.